lib/internal/modules/esm/loader.js

```diff
@@ -107,6 +108,13 @@ class Loader {
     }
 
     const { format } = getFormatResponse;
+    if (format === null) {
+      const dataUrl = RegExpPrototypeExec(
+        /^data:([^/]+\/[^;,]+)(?:[^,]*?)(;base64)?,/,
+        url,
+      );
+      throw new ERR_UNKNOWN_MODULE_FORMAT(dataUrl ? dataUrl[1] : 'null');
```

Comment:
I am fairly certain that [`ERR_UNKNOWN_MODULE_FORMAT`](https://nodejs.org/api/errors.html#errors_err_unknown_module_format) is really only supposed to be used in case the module format is not one of the following five supported formats.

| format | Description |
| -- | -- |
| 'builtin' | Load a Node.js builtin module | 
| 'commonjs' | Load a Node.js CommonJS module |
| 'json' | Load a JSON file |
| 'module' | Load an ES module |
| 'wasm' | Load a WebAssembly module |

Refs: https://nodejs.org/api/esm.html#esm_getformat_url_context_defaultgetformat

Reply:
Agreed, also `ERR_UNKNOWN_MODULE_FORMAT` is a `RangeError`, which doesn't really apply here. I'm going with your `ERR_INVALID_MODULE_SPECIFIER` suggestion.